app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java

```diff
@@ -261,12 +267,55 @@ public void updateListAfterEdit(Habit editedHabit, int pos) {
     // add to list fragment class once swipe is fixed in habit events
     public void updateListAfterDelete(int position) {
         Habit habitToDelete = _habitItemAdapter._snapshots.get(position);
+
+        deleteHabitEvents(_user.getId(), habitToDelete);
+
         _habitList.deleteHabit(getActivity(),
                 _user.getId(),
                 habitToDelete,
                 position);
     }
 
+    /**
+     * Deletes all child habit events of a habit
+     * @param userId (String) The current user's username
+     * @param parentHabit (Habit) The habit to delete
+     */
+    public void deleteHabitEvents(String userId, Habit parentHabit) {
```

Comment:
Minor: If this method is going to be used only within this class, then it should be private.

Reply:
Good catch!